core/block_validator.go

```diff
@@ -163,6 +169,67 @@ func (v *BlockValidator) SanityCheckWorkObjectBlockViewBody(wo *types.WorkObject
 	return nil
 }
 
+func (v *BlockValidator) ApplyPoWFilter(wo *types.WorkObject) pubsub.ValidationResult {
+	var err error
+	powhash, exists := v.hc.powHashCache.Peek(wo.Hash())
+	if !exists {
+		powhash, err = v.engine.VerifySeal(wo.WorkObjectHeader())
+		if err != nil {
+			return pubsub.ValidationReject
+		}
+		v.hc.powHashCache.Add(wo.Hash(), powhash)
+	}
+	newBlockIntrinsic := v.engine.IntrinsicLogS(powhash)
+
+	// cannot have a pow filter when the current header is genesis
+	if v.hc.IsGenesisHash(v.hc.CurrentHeader().Hash()) {
+		return pubsub.ValidationAccept
```

Comment:
Technically this is not true in the chain expansion, correct? Because the genesis block is mined

Reply:
this is just there so that we dont crash, if we are at genesis (just started the node), we cant perform a work filter